src/antsibull_core/schemas/collection_meta.py

```diff
@@ -80,60 +90,135 @@ def _check_reason_text(self) -> Self:
                 )
         return self
 
+    @p.model_validator(mode="after")  # pyre-ignore[56]
+    def _check_renamed_base(self) -> Self:
+        if self.reason == "renamed":
+            if self.new_name is None:
+                raise ValueError("new_name must be provided if reason is 'renamed'")
+        else:
+            if self.new_name is not None:
+                raise ValueError(
+                    "new_name must not be provided if reason is not 'renamed'"
+                )
+            if self.redirect_replacement_major_version is not None:
+                raise ValueError(
+                    "redirect_replacement_major_version must not be provided"
+                    " if reason is not 'renamed'"
+                )
+        return self
+
+
+class RemovalInformation(BaseRemovalInformation):
+    """
+    Stores metadata on when and why a collection will get removed.
+    """
+
+    # The Ansible major version from which the collection will be removed.
+    major_version: t.Union[int, t.Literal["TBD"]]
+
+    @p.model_validator(mode="after")  # pyre-ignore[56]
+    def _check_renamed(self) -> Self:
+        if self.reason == "renamed":
+            if (
+                self.redirect_replacement_major_version is not None
+                and self.major_version != "TBD"
+                and self.redirect_replacement_major_version
+                >= self.major_version  # pyre-ignore[58]
+            ):
+                raise ValueError(
+                    "redirect_replacement_major_version must be smaller than major_version"
+                )
+        else:
+            if self.major_version == "TBD":
+                raise ValueError(
+                    "major_version must not be TBD if reason is not 'renamed'"
+                )
+        return self
+
+
+class RemovedRemovalInformation(BaseRemovalInformation):
+    """
+    Stores metadata on when and why a collection was removed.
+    """
+
+    # The exact version from which the collection has been removed.
+    # This is needed for changelog generation.
+    version: PydanticPypiVersion
+
     @p.model_validator(mode="after")  # pyre-ignore[56]
     def _check_reason_is_renamed(self) -> Self:
         if self.reason != "renamed":
             return self
-        if self.new_name is None:
-            raise ValueError("new_name must be provided if reason is 'renamed'")
         if (
             self.redirect_replacement_major_version is not None
-            and self.major_version != "TBD"
-            and self.redirect_replacement_major_version
-            >= self.major_version  # pyre-ignore[58]
+            and self.redirect_replacement_major_version >= self.version.major
         ):
             raise ValueError(
                 "redirect_replacement_major_version must be smaller than major_version"
```

Comment:
This text needs to be adjusted now that the `major_version` field is no longer there. Also, might be good to add a test case for this error.

Reply:
6ab42046fa1d35fee75dc20a38a446139551a3e7